matron/src/hardware/screen.c

```diff
@@ -396,7 +396,38 @@ double *screen_text_extents(const char *s) {
 
 extern void screen_export_png(const char *s) {
     CHECK_CR
-    cairo_surface_write_to_png(surface, s);
+    static cairo_surface_t *png;
+    static cairo_t *temp; // for bg fill
+    // width = 640 (128*4 pixels with 64 pixel black border)
+    // hieght = 384 (64*4 pixels plus border) 
+    png = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, 640, 384);
```

Comment:
I know I wasn't requested to review this, but I was poking around open PRs anyway. You may be able to use CAIRO_FORMAT_RGB24 and then you wouldn't have to worry about the alpha channel (pixels still fit into a uint32). That could probably allow you to lose the fill area of the code, and the bitwise OR with 0xFF for alpha.

Have you experimented with `cairo_scale()`? You might be able to copy the image data from `surface`, use `create_image_surface_for_data()` with the new dimensions, then apply translate and scale transformations.

Reply:
thanks for taking a look! your suggestions make a ton of sense.

honestly all of my experience with graphics is on raw framebuffers... hence the low-level implementation. i'll see if i can find a minute to try out some of these ideas.